src/isotools/_transcriptome_io.py

```diff
@@ -411,41 +429,60 @@ def add_sample_from_bam(self, fn, sample_name=None, barcode_file=None, fuzzy_jun
                     #        if qual:
                     #            tr['mutations'][sample_name][pos][alt][1].append(qual) #assuming the quality already accounts for cov>1
 
+                    # checks if there is any "clipping" in the alignment, which occurs when part of the read doesn't align to the reference genome
+                    # CIGAR string in a SAM/BAM file represents the operations performed on each aligned segment of the read:
+                    #   '4' corresponds to "soft clipping," which means that the aligned segment is present in the SEQ but does not align to the reference
```

Comment:
I don't think we need to explain what clipping is and how the CIGAR format works here